src/ai/backend/manager/registry.py

```diff
@@ -1550,6 +1563,43 @@ async def _create_kernels_in_one_agent(
                     self.kernel_creation_tracker[kernel_id].cancel()
                     self._post_kernel_creation_infos[kernel_id].set_exception(e)
                 await asyncio.gather(*post_tasks, return_exceptions=True)
+                async with self.db.begin() as conn:
```

Comment:
All writing db transactions must be wrapped as a function and `execute_with_retry()`.  Could we reuse `set_kernel_status()` here?